tests/functionality/scopflow/selfcheck.cpp

```diff
@@ -104,17 +108,28 @@ struct ScopflowFunctionalityTestParameters {
 
 struct ScopflowFunctionalityTests
     : public FunctionalityTestContext<ScopflowFunctionalityTestParameters> {
-  ScopflowFunctionalityTests(std::string testsuite_filename,
-                             int logging_verbosity = EXAGO_LOG_INFO)
-      : FunctionalityTestContext(testsuite_filename, logging_verbosity) {}
 
   using Params = ScopflowFunctionalityTestParameters;
+  MPI_Comm comm;
+  int nprocs;
+
+  ScopflowFunctionalityTests(std::string testsuite_filename, MPI_Comm comm,
+                             int logging_verbosity = EXAGO_LOG_INFO)
+      : FunctionalityTestContext(testsuite_filename, logging_verbosity),
+        comm{comm} {
+    auto err = MPI_Comm_size(comm, &nprocs);
```

Comment:
Same here - we don't want to repeat ourselves if possible

Reply:
Not sure what you are referring to here. What is repeating?

Reply:
The call to get rank and the associated error checking is repeated.

Maybe not worth the effort, but I was thinking there could be a quick way to move this to be common code across all tests, and not code repeated in each tests source files.

Reply:
I can add a utility function that does this. It would be slicker if the FunctionalityTestContext was MPI-aware, but that would be a bigger lift.

Reply:
Okay, I've made the changes and rebased against develop. The checks are still in main, but I got rid of them everywhere else. If this passes then I'm good.